### PR TITLE
Removing un-needed packages

### PR DESCRIPTION
**Pull request by kesne**
I noticed this package was significantly larger than I was expecting, and it seems like there's a bunch of prod dependencies that aren't needed. This just cleans them up.

### CODE REVIEW

**Comment by macklinu** [comment]
This PR has merge conflicts

**Comment by rogeliog** [comment]
Just some merge conflicts... sorry for the long response

File: package.json [hunk 1]
```diff
@@ -19,19 +19,11 @@
...
-    "babel-polyfill": "^6.26.0",
-    "babel-register": "6.26.0",
-    "jest-worker": "21.3.0-beta.2",
-    "lodash": "4.17.4",
-    "minimatch": "3.0.4",
-    "mocha": "3.5.0",
-    "pify": "3.0.0",
-    "throat": "4.1.0",
-    "worker-farm": "1.5.0"
+    "jest-worker": "^21.2.1"
```
Comment (SimenB) [inline package.json]: can you bump this to 22?

**Comment by SimenB** [comment]
Rebased this, and bumped to jest-worker@22

File: integrationTests/__snapshots__/failing.test.js.snap [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 exports[`Works when it has failing tests 1`] = `
 "FAIL integrationTests/__fixtures__/failing/__src__/file1.js
-Company policies require ⚔️ 🏃 in every file
```
Comment (SimenB) [inline integrationTests/__snapshots__/failing.test.js.snap]: not sure about this one. Change in jest 22

Comment (rogeliog) [reply integrationTests/__snapshots__/failing.test.js.snap]: Not sure that I understand why that happened

Comment (SimenB) [reply integrationTests/__snapshots__/failing.test.js.snap]: Tracked this down, it's https://github.com/facebook/jest/pull/4504 (which was released in jest 21.2.0)

**Comment by SimenB** [comment]
I've left jest at version 21 for the node 4 support. Can drop it in a separate PR if wanted

**Pull request merged**